bids/layout/layout.py

```diff
@@ -353,25 +358,35 @@ def do_begin(conn):
 
         self.session = sa.orm.sessionmaker(bind=engine)()
 
-    def _make_db_paths(self, database_path):
+    @staticmethod
+    def _make_db_paths(database_path):
         if database_path is not None:
-            database_file = os.path.join(database_path, 'layout_index.sqlilte')
-            database_sidecar = os.path.join(database_path, 'layout_args.json')
-            os.makedirs(database_path, exist_ok=True)
+            database_path = Path(database_path)
+            database_file = database_path / 'layout_index.sqlilte'
+            database_sidecar = database_path / 'layout_args.json'
+            database_path.mkdir(exist_ok=True)
         else:
             database_file = None
             database_sidecar = None
         return database_file, database_sidecar
 
-    def _sanitize_instance_args(self):
-        instance_args = {
-            a: self.__dict__[a] for a in
-            ['root', 'validate', 'absolute_paths', 'regex_search', 'config_file']
-            if a in self.__dict__
-        }
+    def _sanitize_init_args(self, **kwargs):
+        """ Clean up initalization arguments for saving. """
+        self.init_args = kwargs
+        # Make ignore and force_index hashable
         for k in ['ignore', 'force_index']:
-            instance_args[k] = [str(a) for a in self.__dict__[k]]
-        return instance_args
+            self.init_args[k] = [
+                str(a) for a in self.init_args[k] if a is not None] \
+              if self.init_args[k] is not None else None
```

Comment:
Hashable? Should it be a `tuple`, then?

Reply:
It's because `re` object can be passed in.

Reply:
Ack, but then we should probably reload them as `re` objects

Reply:
Hmm, given that this is won't affect the index, I'm tempted to say just leave it as a string. It's an edge case.

Reply:
Oh, sorry, I think there was a crossed wire here. I meant that if `init_args` needs to be hashable, then this should be `tuple(str(a) ...)` instead of `[str(a) ...]`. But I'm not really sure about this whole hashability thing.

If this works, it's fine.